spec/controllers/categories_controller_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'spec_helper'
+require 'shoulda-matchers'
+
+include Testable
+
+describe CategoriesController do
+  it { should use_before_action :require_login }
+
+  describe 'GET show' do
+    before do
+      test_login
+      #the web call below requires an 'id', but I pass a slug into this field by overriding to_param w/ the sluggable gem that I made in course 2.
+      get :show, id: Fabricate(:category).slug
```

Comment:
Thanks to `to_param` you shouldn't need to call `slug`, you should be able to just pass in the object.